lib/rules/max-empty-lines/index.js

```diff
@@ -38,60 +39,65 @@ const rule = function(max, options) {
       return;
     }
 
-    const repeatLFNewLines = _.repeat("\n", maxAdjacentNewlines);
-    const repeatCRLFNewLines = _.repeat("\r\n", maxAdjacentNewlines);
     const ignoreComments = optionsMatches(options, "ignore", "comments");
 
     eachRoot(root, checkRoot);
 
-    // We must check comments separately in order to accommodate stupid
-    // `//`-comments from SCSS, which postcss-scss converts to `/* ... */`,
-    // which adds to extra characters at the end, which messes up our
-    // warning position
-    if (!ignoreComments) {
-      root.walkComments(comment => {
-        const source =
-          (comment.raws.left || "") + comment.text + (comment.raws.right || "");
-        styleSearch({ source, target: "\n" }, match => {
-          checkMatch(source, match.endIndex, comment, 2);
-        });
-      });
-    }
```

Comment:
Is the above section supposed to have been removed?

Won't the warning position for SCSS be inaccurate with this removed?

Reply:
@ntwb Well you found something right that there will be false warning position in non-standard syntaxes but it's not related to what you pointed above. Because above is just a part which looks for comments in input css string based on `ignore: ['comments']` option which I already did after few lines later. 

Beside that problem is `root.toString() `. Because it cannot be used to check css string. Instead of you have to use PostCSS' node functions to achieve this.

Then there is one more problem. It's same problem why I couldn't in my first PR and also now in my second PR because PostCSS' node tree structure is so crap.  

For example, in node tree, `before` is the space symbols before the node. It also stores * and _ symbols before the declaration (IE hack), `after` is the space symbols after the last child of the node to the end of the node, they say. WHY?

```
/* before goes here */
.a {
  color: #fff;
  /* after goes here */
}
```

Seriously WHY? Why before is outter but after is inner? Maybe it's my personal thought but this is how I think. So for final words, don't wait for me if you think you can fix this. You are %100 free to edit my PR to fix this solution. Only thing I know that this bug stands for over one year and has to be fixed.